### PR TITLE
[8.x] Create Collection `chunkInto` method

### PR DESCRIPTION
**Pull request by browner12**
This methods defines the number of chunks we would like to end up with. It fills all non-last chunks first, before placing the remainder in the final chunk.

This method is similar to, but different than, `split()`.

Given the following Collection:

```php
$array = collect([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]);
```

`$array->split(3)` will return 3 chunks of sizes 4, 3, and 3.

`$array->chunkInto(3)` will return 3 chunks of sizes 4, 4, and 2.

### CODE REVIEW

**Comment by taylorotwell** [comment]
Can you explain in simple terms how it is fundamentally different than `split` and when you would use one over the other?

**Comment by browner12** [comment]
`split` will make it so each chunk never differs in count from another chunk by more than 1. 

Screen Shot 2020-11-20 at 9 37 16 AM

`chunkInto` will fill all earlier chunks completely before allocating the remainder to the final chunk.

Screen Shot 2020-11-20 at 9 37 10 AM

On one of our sites, we have a sitemap page with a list of all the pages on the site. I wanted to display them in columns, and I wanted it display like in the second picture.  This is where `chunkInto` handles what I need, and `split` does not.

**Comment by taylorotwell** [comment]
One thing giving me a bit of pause is we have used the "into" language to indicate something will be injected into a class... mapInto and pipeInto for example.

**Comment by browner12** [comment]
Gotcha. Yah, I'm totally open to other names. Maybe sticking with the `split` prefix is more appropriate, too, since `chunk` defines the number of items **in** the chunk, and `split` defines the number **of** chunks. This new method just allots them differently.

- `splitBy`
- `splitIn`

Another option would be to pass a second parameter flag to `split()` although I know we're not normally too keen on that.

**Comment by taylorotwell** [comment]
I think I'm fine with `splitIn`.

File: src/Illuminate/Collections/LazyCollection.php [hunk 1]
```diff
@@ -1057,6 +1057,17 @@ public function chunk($size)
...
 
+    /**
+     * Split a collection into a certain number of groups, and fill the first groups completely.
+     *
+     * @param  int  $numberOfGroups
+     * @return static
+     */
+    public function splitIn($numberOfGroups)
+    {
+        return $this->chunk(ceil($this->count() / $numberOfGroups));
```
Comment (JosephSilber) [inline src/Illuminate/Collections/LazyCollection.php:1068]: This enumerates the whole collection twice :cry:

Comment (browner12) [reply src/Illuminate/Collections/LazyCollection.php:1068]: I'll admit I don't understand the lazy collections as well, so if there's a better way to do this, definitely open to it.

**Pull request merged**